### PR TITLE
Use Allowlist/Denylist instead of Whitelist/Blacklist

### PR DESCRIPTION
**Pull request by tanishiking**
Resolve https://github.com/scala-native/scala-native/issues/3534

https://www.splunk.com/en_us/blog/learn/blacklist-whitelist-inclusivity.html
https://docs.scala-lang.org/contribute/inclusive-language-guide.html

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -392,26 +392,26 @@ object Settings {
...
+    if (nonexistentDenylisted.nonEmpty) {
       throw new AssertionError(
-        s"Sources not found for blacklisted tests:\n$nonexistentBlacklisted"
+        s"Sources not found for denylisted tests:\n$nonexistentDenylisted"
       )
     }
   }
 
-  def sharedTestSource(withBlacklist: Boolean) = Def.settings(
+  def sharedTestSource(withDenylist: Boolean) = Def.settings(
```
Comment (tanishiking) [inline project/Settings.scala:409]: These are used only in a scala-native project, and it shouldn't affect users APIs

File: scala-partest-tests/src/test/resources/scala/tools/partest/scalanative/2.12.18/BlacklistedTests.txt [hunk 2]
```diff
@@ -1,1089 +0,0 @@
...
-run/t4300.scala
-run/t3361.scala
-run/t8017
-run/t8334.scala
-run/t8803.scala
-run/t9697.scala
-run/t10290.scala
-
-## Other
-run/richs.scala
```
Comment (tanishiking) [inline scala-partest-tests/src/test/resources/scala/tools/partest/scalanative/2.12.18/BlacklistedTests.txt:1089]: Not sure why github doesn't recognize this change as rename 🤔

**Review by WojciechMazur — COMMENTED** [review]
LGTM, the failing tests are probably not related to this change, would restart them later

**Comment by LeeTibbert** [comment]
This is PR is a definite move forward, thank you for getting it done.

**Pull request merged**